### PR TITLE
[1.5] backport adding grafanaInClusterURL in Kiali helm chart

### PR DESCRIPTION
**Pull request by jotak**
See also https://github.com/istio/istio/pull/21255

### CODE REVIEW

**Comment by jotak** [comment]
Also removing hardcoded "istio-system" in URL as it's not necessary and would reduce number of default working cases.

**Comment by jotak** [comment]
Also removing hardcoded "istio-system" in URL as it's not necessary and would reduce number of default working cases.

**Comment by fpesce** [comment]
@jotak I think you need to run `make gen` on this one.

Also, I'm not seeing the original PR on master,  is that a cherry pick ?

**Comment by jotak** [comment]
@fpesce it's not merged in master yet, it's here: https://github.com/istio/istio/pull/21255
It's not exactly a cherry-pick: in 1.5 we already added jaegerInCluserURL ; then in master jaegerInCluserURL and grafanaInCluserURL were added by another user that was merged almost at the same time ... so now I'm trying to make all of these consistent both in master and 1.5, by adding grafanaInCluserURL here.

About `make gen`, yes for some reason it doesn't worked on my machine (working well on master but not in 1.5) so I ran `make operator-proto` hoping that would be sufficient. Looks like it isn't...

**Comment by sdake** [comment]
> @fpesce it's not merged in master yet, it's here: #21255
> It's not exactly a cherry-pick: in 1.5 we already added jaegerInCluserURL ; then in master jaegerInCluserURL and grafanaInCluserURL were added by another user that was merged almost at the same time ... so now I'm trying to make all of these consistent both in master and 1.5, by adding grafanaInCluserURL here.
> 

@fpesce for some reason, a PR was merged into 1.5 before being merged into master, creating an inconsistent state in the protobufs between master and release 1.5. This should fix that problem.

> About `make gen`, yes for some reason it doesn't worked on my machine (working well on master but not in 1.5) so I ran `make operator-proto` hoping that would be sufficient. Looks like it isn't...

@jotak we do not use the BUILD_WITH_CONTAINER=1 setting for 1.5 branch. We decided it would be better to enable this after the branch. On reflection, we should have BUILD_WITH_CONTAINER=1 on the release-1.5 branch. It is far too late to change that now though, so we will struggle a bit with this problem for install backports for 1.5 until 1.5 is released.

I'd recommend running `BUILD_WITH_CONTAINER=1 make gen` to work around the problem.

**Comment by sdake** [comment]
TL;DR new implementation was directly merged to a release branch without a proper exception.

@fpesce IMO The correct solution involves the following actions:

1. revert https://github.com/istio/istio/pull/21256
1. backport https://github.com/istio/istio/pull/21393
1. backport https://github.com/istio/istio/pull/21255

For more rationale on why the revert is needed, see: https://github.com/istio/istio/pull/21256#issuecomment-593030058

Other approaches can be taken as long as the protobuf is consistent and master is the source of truth for new implementation work.

**Comment by fpesce** [comment]
@sdake I did not realize that these PRs were not in master.

I'll check the impact of reverting something merged 5 days ago. Hoping it'll be doable.

**Comment by fpesce** [comment]
@sdake Also note that master has seen a lot of changes, so if this manual PR is the only way to cherrypick the mentioned changes, we should accept it.

**Comment by jotak** [comment]
@fpesce @sdake looks like protobuf isn't fixed yet between 1.5 and master:

```
$ git diff release-1.5 master -- operator/pkg/apis/istio/v1alpha1/values_types.proto

(...)
   TypeMapStringInterface auth = 7;
 
-  string jaegerInClusterURL = 8;
+  string grafanaInClusterURL = 8;
 
-  string grafanaInClusterURL = 9;
+  string jaegerInClusterURL = 9;

(...)
```

I'm going to fix it.

**Comment by jotak** [comment]
This time codegen seemed to work. Not sure what was wrong last time when it didn't, basically many generated files were quite suspicious with changes where I wouldn't expect them.

**Comment by sdake** [comment]
> @fpesce @sdake looks like protobuf isn't fixed yet between 1.5 (this PR) and master:
> 
> ```
> $ git diff release-1.5 master -- operator/pkg/apis/istio/v1alpha1/values_types.proto
> 
> (...)
>    TypeMapStringInterface auth = 7;
>  
> -  string jaegerInClusterURL = 8;
> +  string grafanaInClusterURL = 8;
>  
> -  string grafanaInClusterURL = 9;
> +  string jaegerInClusterURL = 9;
> 
> (...)
> ```
> 
> I'm going to fix it.
> [edit: done]

@jotak thanks again for your work. In the future, please submit to master first, such that the backport process is smoother. When submitting to a release branch, it makes it harder for the maintainers to understand what has been done. Can you link your PR for making master consistent with 1.5?

Cheers
-steve

**Comment by dgn** [comment]
> > @fpesce @sdake looks like protobuf isn't fixed yet between 1.5 (this PR) and master:
> > ```
> > $ git diff release-1.5 master -- operator/pkg/apis/istio/v1alpha1/values_types.proto
> > 
> > (...)
> >    TypeMapStringInterface auth = 7;
> >  
> > -  string jaegerInClusterURL = 8;
> > +  string grafanaInClusterURL = 8;
> >  
> > -  string grafanaInClusterURL = 9;
> > +  string jaegerInClusterURL = 9;
> > 
> > (...)
> > ```
> > 
> > 
> > I'm going to fix it.
> > [edit: done]
> 
> @jotak thanks again for your work. In the future, please submit to master first, such that the backport process is smoother. When submitting to a release branch, it makes it harder for the maintainers to understand what has been done. Can you link your PR for making master consistent with 1.5?
> 
> Cheers
> -steve

Hi Steve, I double-checked before approving.. Checked again now:
```
17:36:48 localhost (release-1.5) ~/dev/go/src/istio.io/istio$ git diff release-1.5 master -- operator/pkg/apis/istio/v1alpha1/values_types.proto
diff --git a/operator/pkg/apis/istio/v1alpha1/values_types.proto b/operator/pkg/apis/istio/v1alpha1/values_types.proto
index 726e15e85..0e0c30243 100644
--- a/operator/pkg/apis/istio/v1alpha1/values_types.proto
+++ b/operator/pkg/apis/istio/v1alpha1/values_types.proto
@@ -651,7 +651,8 @@ message STSConfig {
 
 message IstiodConfig {
   // If enabled, all control plane functionality will be handled by a single deployment.
-  google.protobuf.BoolValue enabled = 1;
+  // Deprecated. This field is always ignored and assumed true.
+  google.protobuf.BoolValue enabled = 1 [deprecated=true];
 }
 
 // GlobalLoggingConfig specifies the global logging level settings for the Istio control plane components.
@@ -2203,7 +2204,7 @@ message Values {
 
   MixerConfig mixer = 8;
 
-  NodeAgentConfig nodeagent = 9;
+  NodeAgentConfig nodeagent = 9 [deprecated=true];
 
   PilotConfig pilot = 10;
 

```

Are you sure you pulled?

**Comment by jotak** [comment]
> @jotak thanks again for your work. In the future, please submit to master first, such that the backport process is smoother. When submitting to a release branch, it makes it harder for the maintainers to understand what has been done. Can you link your PR for making master consistent with 1.5?
> 
> Cheers
> -steve

I don't have any more PR to master, consistency was achieved here through this PR (unless I'm missing something).

**Comment by voroniys** [comment]
@sdake can you take care that this PR will be included in release 1.5 ? Currently only PR which adds jagerInClusterURL included in 1.5.0-beta.5 what is actually not enough.

**Comment by dgn** [comment]
> @sdake can you take care that this PR will be included in release 1.5 ? Currently only PR which adds jagerInClusterURL included in 1.5.0-beta.5 what is actually not enough.

Hi @voroniys, this change will be in 1.5.0 - it's already on the release branch. It just didn't make the cutoff for the beta.5

**Comment by sdake** [comment]
@dgn I think this PR didn't make the cutoff for 1.5 but should be in 1.5.1. Can you confirm?

Cheers
-steve

**Comment by voroniys** [comment]
I've testet it with 1.5 - seems to be working.

**Comment by dgn** [comment]
@sdake No, this was included with 1.5.0- it was merged on Monday, we built the release on Wednesday

@voroniys thanks for testing
